### ISSUE TITLE
Support alternative multi-selection model for a WAI-ARIA ListBox

### ISSUE THREAD

**Issue by rhmoller** [adobe/react-spectrum#1638]
# 🙋 Feature Request

Add support for the alternative multi selection model where selection will be replaced unless a modifier key is pressed. See "Alternative selection model" in https://www.w3.org/TR/wai-aria-practices-1.1/#Listbox

## 🤔 Expected Behavior

When the user presses an item, all other items should be deselected unless a modifier key is pressed (shift, ctrl/command).

## 😯 Current Behavior

Currently the selection manager toggles selection on click in multi selection mode.

## 💁 Possible Solution

In line 343 of `SelectionManager.ts` `toggleSelection` is invoked.  https://github.com/adobe/react-spectrum/blob/main/packages/%40react-stately/selection/src/SelectionManager.ts#L343

If the selection manager is running in the alternative multi selection mode it should only call `toggleSelection` if `ctrlKey`/`metaKey` is pressed, otherwise it should call `replaceSelection`

There should be a way to indicate that the alternative selection model is desired. Maybe as an additional property on `MultipleSelection` https://github.com/adobe/react-spectrum/blob/main/packages/%40react-types/shared/src/selection.d.ts#L28 or more drastically as an additonal value for the `SelectionMode` string literal type.

It should be possible to pass this flag all the way through higher level hooks like `useListState()` etc.

## 🔦 Context

One use case is a master-detail view where the master is a list box.
When the user clicks an item, they will expect only that item to be selected and displayed in the detail pane.
Multiple items can be selected by shift-clicking or ctrl-clicking, in which case the detail pane will display possible operations to perform on the items.

## 💻 Examples

An example of this behaviour can be seen in the web version of Outlook in Office365

## 🧢 Your Company/Team

Stibo Systems

## 🎁 Tracking Ticket (optional)

N/A

**Comment by devongovett** [comment]
Yep, we will need this selection model for some upcoming components as well. One potential issue is that it doesn't work well on touch devices without modifier keys, as there would be no way to select multiple. We will need to figure out an alternative there.
